src/bejeweled/board/GemFactory.java

```diff
@@ -0,0 +1,17 @@
+package bejeweled.board;
+
+public class GemFactory {
+	
+	public Gem createGem(int row, int col, GemType type, SpecialType special) {
+		Gem gem = null;
```

Comment:
If you just return the newly made DoublePointGem or DeleteRowGem you dont need to make a Gem gem variable.